src/clj/game/cards/resources.clj

```diff
@@ -1896,12 +1922,14 @@
    "Synthetic Blood"
    {:events {:damage {:req (req (genetics-trigger? state side :damage))
                       :msg "draw 1 card"
-                      :effect (effect (draw :runner))}}}
+                      :async true
+                      :effect (effect (draw :runner eid 1 nil))}}}
 
    "Tallie Perrault"
    {:abilities [{:label "Draw 1 card for each Corp bad publicity"
-                 :effect (effect (trash card {:cause :ability-cost})
-                                 (draw (+ (:bad-publicity corp) (:has-bad-pub corp))))
+                 :async true
+                 :effect (req (wait-for (trash state side card {:cause :ability-cost})
+                                        (draw state side eid (+ (:bad-publicity corp) (:has-bad-pub corp)) nil)))
```

Comment:
We really need to clean up bad publicity, lol. This is ugly af

Reply:
Only fitting ;)